### DISCUSSION TITLE
How can I force the window to keep fullscreen?

### DISCUSSION THREAD

**Discussion by Mr-Milk** [tauri-apps/tauri#3313] [Q&A]
If I don't want my user to exit fullscreen by pressing keyboard `ESC` or `F11` and by switching between desktop. Is there any way I could achieve this in tauri?

**Comment by seanmamasde** [comment]
hello, I think I accidentally found (part of) the answer, though I am pretty unsure. I am creating a windows screensaver app, and screensavers on windows are essentially executables running fullscreen with `.scr` extension. I only have only have `"fullscreen": true` and `"title"` property under `tauri.conf.json > tauri > windows[]`. The result of this is that I can't use either `f11` or `esc` to exit fullscreen. Do note that I only have windows machine tested currently. If you are on other platforms, you can try it out yourself to verify my claim.

**Reply by mahadi-hossain** [reply]
It works on linux too.

**Reply by mgrisole** [reply]
Thank you, on macOS, it does open in fullscreen mode but it leaves when pressing `esc`.
The only workaround I found so far is by adding a global event listener:

```js
    const onKeyDown = (event: KeyboardEvent) => {
      if (event.key !== "Escape") {
        return;
      }

      event.preventDefault();
    };

    window.addEventListener("keydown", onKeyDown);
```

I am all ears for a more elegant solution though.

**Comment by Silbad** [comment]
Hi,
In js you can do that : 
```js
import { appWindow } from "@tauri-apps/api/window";
await appWindow.setFullscreen(true);
```
In tauri.conf.json : 
```json
    "tauri": {
        "allowlist": {
            "window": {
                "setFullscreen": true
            }
        },
```

**Reply by StarKnightt** [reply]
Thanks brother, appreciate it.

**Reply by alvinometric** [reply]
With Tauri v2:

JavaScript
```js
import { getCurrentWindow } from '@tauri-apps/api/window';
await getCurrentWindow().setFullscreen(true);
```

capabilities.json
```json
"permissions": [
  "core:window:allow-set-fullscreen"
]
```
